src/ai/backend/manager/registry.py

```diff
@@ -192,6 +194,54 @@
 SESSION_NAME_LEN_LIMIT = 10
 
 
+async def get_known_registries(
+    db: ExtendedAsyncSAEngine,
+) -> Mapping[str, Mapping[str, yarl.URL]]:
+    async with db.begin_readonly_session() as db_session:
+        registries: list[tuple[str, str, str]] = (
+            await db_session.execute(
+                sa.select([
+                    ContainerRegistryRow.project,
+                    ContainerRegistryRow.registry_name,
+                    ContainerRegistryRow.url,
+                ])
+            )
+        ).fetchall()
+
+        result: MutableMapping[str, MutableMapping[str, yarl.URL]] = {}
+
+        for item in registries:
+            project, registry_name, url = item
+
+            if project not in result:
+                result[item[0]] = {}
+
+            result[project][registry_name] = yarl.URL(url)
+
+        return result
+
+
+async def get_registry_info(db: ExtendedAsyncSAEngine, registry_id: str) -> tuple[yarl.URL, dict]:
+    async with db.begin_readonly_session() as db_session:
+        result: tuple[str, Optional[str], Optional[str]] = (
+            await db_session.execute(
+                sa.select([
+                    ContainerRegistryRow.url,
+                    ContainerRegistryRow.username,
+                    ContainerRegistryRow.password,
+                ]).where(ContainerRegistryRow.id == registry_id)
+            )
+        ).fetchall()[0]
+
+        if not result:
+            raise UnknownImageRegistry(registry_id)
+
+        url, username, password = result
+        creds = {"username": username, "password": password}
+
+        return yarl.URL(url), creds
```

Comment:
How about locating those functions in `models/container_registry.py` file? just a question!

Reply:
These functions are only used in this file, so I defined them here.

If we move these function, we need to update `test_handle_heartbeat`.

Reply:
then how about renaming the functions to specify that they get data about container registry?
it would be confusing because this file is `registry.py` and we cannot understand what `registry` means in those name at the first glance